src/app/config/config.c

```diff
@@ -3010,9 +3031,16 @@ is_local_addr, (const tor_addr_t *addr))
      * it was on net 0, which is already done by tor_addr_is_internal.
      */
     if ((last_resolved_addr & (uint32_t)0xffffff00ul)
-        == (ip & (uint32_t)0xffffff00ul))
+        == (addr_ipv4 & (uint32_t)0xffffff00ul))
```

Comment:
This code is correct, please don't change it.

Instead, replace last_resolved_addr with addr_ipv4